Use 5 bits to encode setting words

fiam opened the pull request:
Saves 1672 bits on F3 without any lost functionality. Retriving a
setting name is significantly slower, but still not slow enough to be
problematic.

fiam committed:
Use 5 bits to encode setting words

Saves 1672 bits on F3 without any lost functionality. Retriving a
setting name is significantly slower, but still not slow enough to be
problematic.

alexbirkett commented on src/utils/settings.rb:
```diff
@@ -455,13 +467,62 @@ def write_impl_file(file)
+            else
+                idx = symbols.index(c)
+                if idx.nil?
+                    if rem_symbols == 0
```
Is `rem_symbols` ever decremented?

fiam replied:
Nice catch! That’s definitely a bug, I’ll fix it.

fiam committed:
Decrement rem_symbols when a symbol slot is used

Spotted by @alexbirkett

fiam committed:
Output the number of bits used for encoding characters from settings.rb

Instead of hardcoding the number of bits in both settings.c and
settings.rb, make the latter output a constant to the
settings_generated.h file.
